aiservices/vertexai_provider.go

```diff
@@ -0,0 +1,42 @@
+package aiservices
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+)
+
+type googleVertexAIModelProvider struct {
+	config VertexAIModelConfig
+}
+
+func NewGoogleVertexAIModelProvider(config VertexAIModelConfig) (ModelProvider, error) {
+	return &googleVertexAIModelProvider{config: config}, nil
+}
+
+func (g googleVertexAIModelProvider) GetFastModel() (Model, error) {
+	fastModel, err := NewGoogleVertexAIFastModel(context.Background(), g.config)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create vertex ai fast model")
+	}
+	return fastModel, nil
+}
+
+func (g googleVertexAIModelProvider) GetReasoningModel() (Model, error) {
+	reasoningModel, err := NewGoogleVertexAIReasoningModel(context.Background(), g.config)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create vertex ai fast model")
+	}
+	return reasoningModel, nil
+}
+
+func (g googleVertexAIModelProvider) GetModelByID(s string) (Model, error) {
+	switch s {
+	case vertexAIFastModelId:
```

Comment:
This is wrong. The intent is to find a model by ID defined in https://models.dev/

You can just return error saying that `GetModelByID` is not supported for this provider